features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/TimeseriesPersisterFactory.java

```diff
@@ -44,25 +44,46 @@
 import org.opennms.netmgt.rrd.RrdRepository;
 
 import com.codahale.metrics.MetricRegistry;
+import org.opennms.netmgt.timeseries.TimeseriesStorageManager;
+import org.opennms.netmgt.timeseries.stats.StatisticsCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
 /**
  * Factory for {@link TimeseriesPersister}.
  *
  * @author jwhite
  */
 public class TimeseriesPersisterFactory implements PersisterFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(TimeseriesPersisterFactory.class);
 
     private final TimeseriesWriter timeseriesWriter;
     private final MetaTagDataLoader metaTagDataLoader;
     private final Cache<ResourcePath, Set<Tag>> configuredAdditionalMetaTagCache;
     private final MetricRegistry registry;
 
     @Inject
-    public TimeseriesPersisterFactory(final TimeseriesWriter timeseriesWriter,
-                                      final MetaTagDataLoader metaTagDataLoader,
+    public TimeseriesPersisterFactory(final MetaTagDataLoader metaTagDataLoader,
+                                      final StatisticsCollector stats,
+                                      @Named("timeseriesStorageManager") final TimeseriesStorageManager timeseriesStorageManager,
                                       @Named("timeseriesPersisterMetaTagCache") final CacheConfig cacheConfig,
-                                      @Named("timeseriesMetricRegistry") MetricRegistry registry) {
-        this.timeseriesWriter = timeseriesWriter;
+                                      @Named("timeseriesMetricRegistry") MetricRegistry registry,
+                                      @Named("timeseries.ring_buffer_size") Integer ringBufferSize,
+                                      @Named("timeseries.writer_threads") Integer numWriterThreads,
+                                      @Named("timeseries.backend") String backend,
+                                      @Named("timeseries.offheap.batchSize") final int batchSize,
+                                      @Named("timeseries.offheap.path") final String path,
+                                      @Named("timeseries.offheap.maxFileSize") final Long maxFileSize) {
+        if (OffheapTimeSeriesWriter.OFFHEAP_NAME.equals(backend)) {
```

Comment:
I wonder if we shouldn't use a more precise name for that parameter, e.g.
_timeseries.writer.buffer_ instead of _timeseries.backend_

I would then maybe also rename the other parameter to like:
_timeseries.writer.buffer.size_ instead of _timeseries.ring_buffer_size_

The goal would be to make it more clear what these parameters are.

Reply:
timeseries.writer.buffer is fine for me too. May be backend is a bit misleading.

keeping timeseries.ring_buffer_size just because i don't want to have duplicate buffer size param, also I don't want to force existing user change the config. Do you know any way in xml to fallback to timeseries.ring_buffer_size if timeseries.writer.buffer.size not found ?